iib/workers/tasks/build_regenerate_bundle.py

```diff
@@ -367,14 +379,55 @@ def _resolve_image_pull_specs(bundle_metadata, labels, pinned_by_iib, registry_r
             log.debug('%s will be replaced with %s', pullspec, new_pullspec.to_str())
             replacement_pullspecs[pullspec] = new_pullspec
 
+    _replace_csv_pullspecs(bundle_metadata, replacement_pullspecs)
+
+
+def _replace_image_name_from_labels(bundle_metadata, replacement_template):
+    """
+    Replace repo/image-name in the CSV pull specs with values from their labels.
+
+    :param dict bundle_metadata: the dictionary of CSV's and relatedImages pull specifications
+    :param str replacement_template: the template specifying which label values to use for
+        replacement
+    """
+    replacement_pullspecs = {}
+
+    for pullspec in bundle_metadata['found_pullspecs']:
+        new_pullspec = ImageName.parse(pullspec.to_str())
+        pullspec_labels = get_image_labels(pullspec.to_str())
+        try:
+            modified_namespace_repo = replacement_template.format(**pullspec_labels)
+        except KeyError:
+            raise IIBError(
+                f'Pull spec {pullspec.to_str()} is missing one or more label(s)'
+                f' required in the image_name_from_labels {replacement_template}.'
+                f' Available labels: {", ".join(list(pullspec_labels.keys()))}'
+            )
+        new_pullspec.namespace, new_pullspec.repo = modified_namespace_repo.split('/', 1)
```

Comment:
We also need to take into account flattening of repo names. For example, `spam/eggs/bacon` is invalid in some registries, i.e. quay.io. It should be turned into `spam/eggs-bacon`.

Reply:
In order to flatten the repo name, you will have to use the enclose_repo customization in #245. So I think we can ignore flattening here. Since the customization only promises to replace the image name with labels.

Reply:
I can see how the user will be stranded if the name label has `spam/eggs/bacon` and then they specify `spam` as `namespace` and `enclose_glue` as `-`. It would become `spam/spam-eggs-bacon` which is invalid. In that case, we can alter the other PR to have namespace as optional in which case, the current namespace will be used and or the name, `/`s will be replaced with the enclose glue. thoughts?

Reply:
This should be handled in #245, I agree.